File: AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/MAX/Native Ads/ALMAXAdPlacerTableViewController.m
//
//  ALMAXAdPlacerTableViewController.m
//  AppLovin MAX Demo App - ObjC
//
//  Created by Ritam Sarmah on 4/1/22.
//  Copyright © 2022 AppLovin Corporation. All rights reserved.
//

#import "ALMAXAdPlacerTableViewController.h"
#import <AppLovinSDK/AppLovinSDK.h>

@interface ALMAXAdPlacerTableViewController()<MAAdPlacerDelegate>

@property (nonatomic, strong) MATableViewAdPlacer *adPlacer;
@property (nonatomic, strong) NSArray<NSString *> *data;

@end

@implementation ALMAXAdPlacerTableViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.data = [UIFont.familyNames sortedArrayUsingSelector: @selector(compare:)];
    
    MAAdPlacerSettings *settings = [MAAdPlacerSettings settingsWithAdUnitIdentifier: @"YOUR_AD_UNIT_ID"];
    [settings addFixedPosition: [NSIndexPath indexPathForRow: 2 inSection: 0]];
    [settings addFixedPosition: [NSIndexPath indexPathForRow: 8 inSection: 0]];
    settings.repeatingInterval = 10;
    
    // If using custom views, you must also set the `nativeAdViewNib` and `nativeAdViewBinder` properties on the ad placer
    
    self.adPlacer = [MATableViewAdPlacer placerWithTableView: self.tableView settings: settings];
    self.adPlacer.delegate = self;
    [self.adPlacer loadAds];
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.data.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView al_dequeueReusableCellWithIdentifier: @"ALMAXAdPlacerTableViewCell" forIndexPath: indexPath];
    cell.textLabel.text = self.data[indexPath.row];
    return cell;
}

#pragma mark - MAAdPlacerDelegate

- (void)didLoadAdAtIndexPath:(NSIndexPath *)indexPath {}
- (void)didRemoveAdsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths {}
- (void)didClickAd:(MAAd *)ad {}
- (void)didPayRevenueForAd:(MAAd *)ad {}

@end

File: AppLovin MAX Demo App - Swift/AppLovin MAX Demo App - Swift/MAX/Native Ads/ALMAXAdPlacerTableViewController.swift
//
//  ALMAXAdPlacerTableViewController.swift
//  AppLovin MAX Demo App - Swift
//
//  Created by Ritam Sarmah on 4/1/22.
//  Copyright © 2022 AppLovin. All rights reserved.
//

import UIKit
import AppLovinSDK

class ALMAXAdPlacerTableViewController: UITableViewController
{
    private let data = UIFont.familyNames.sorted()
    
    private var adPlacer: MATableViewAdPlacer!
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        let settings = MAAdPlacerSettings(adUnitIdentifier: "YOUR_AD_UNIT_ID")
        settings.addFixedPosition(IndexPath(row: 2, section: 0))
        settings.addFixedPosition(IndexPath(row: 8, section: 0))
        settings.repeatingInterval = 10;
        
        // If using custom views, you must also set the `nativeAdViewNib` and `nativeAdViewBinder` properties on the ad placer
        
        adPlacer = MATableViewAdPlacer(tableView: tableView, settings: settings)
        adPlacer.delegate = self;
        adPlacer.loadAds()
    }
    
    // MARK: - UITableViewDataSource
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
        return data.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
    {
        let cell = tableView.al_dequeueReusableCell(withIdentifier: "ALMAXAdPlacerTableViewCell", for: indexPath)
        cell.textLabel!.text = data[indexPath.row]
        return cell
    }
}

extension ALMAXAdPlacerTableViewController: MAAdPlacerDelegate
{
    func didLoadAd(at indexPath: IndexPath) {}
    
    func didRemoveAds(at indexPaths: [IndexPath]) {}

    func didClick(_ ad: MAAd) {}
    
    func didPayRevenue(for ad: MAAd) {}
}
